cf_xarray/tests/test_accessor.py

```diff
@@ -935,6 +956,57 @@ def test_get_bounds_dim_name():
     assert mollwds.cf.get_bounds_dim_name("lon") == "bounds"
 
 
+def test_grid_mappings():
+    ds = rotds.copy(deep=False)
+
+    actual = ds.cf.grid_mappings
+    expected = {"air_temperature": ["rotated_pole"], "temp": ["rotated_pole"]}
+    assert ds.cf.grid_mappings == expected
+
+    actual = ds.cf.get_grid_mapping("temp")
+    expected = ds.cf["rotated_pole"]
+    assert_identical(actual, expected)
+
+    actual = ds.cf[["temp"]]
+    assert "rotated_pole" in actual.coords
+
+    # Dataset has grid mappings
+    expected = """\
+    - Grid Mappings:   air_temperature: ['rotated_pole']
+                       temp: ['rotated_pole']
+    """
+    assert dedent(expected) in ds.cf.__repr__()
+
+    # Extraction
+    assert ds.cf["temp"].coords["rotated_pole"] == rotds.rotated_pole
+
+    # DataArray
+    # propagation does not work yet properly
+    actual = ds.cf["temp"].cf.__repr__()
+    assert actual == expected
```

Comment:
> For propagating info when extracting a DataArray, can you add a failing test please? I can take a look then.

I think this should pass, but i am not sure why it's not working properly. The rotated_pole is extracted into coords but no seen in the repr...

Reply:
I think you need to add a specific `grid_mappings` (maybe singular for DataArray) attribute so `ds["temp"].cf.grid_mapping` works.

Reply:
> For bonus points (future PR is OK) I think we should handle the repr for grid_mapping specifically by surfacing extra info like the actual mapping name, and potential other useful info.

I think that would require some adaptations to `make_text_section`, right? I am a little undecided what to do with the repr for grid mappings as a coordinate. `ds["temp"].cf.grid_mapping` now returns a DataArray, no dictionary...

Reply:
#391 makes some big repr changes, so we can just skip this for now.

Reply:
Sorry i meant #409